baxter_collaboration_lib/src/robot_interface/gripper.cpp

```diff
@@ -27,51 +29,39 @@ Gripper::Gripper(std::string _limb, bool _use_robot) :
     init_state.moving     = EndEffectorState::STATE_UNKNOWN;
 
     setGripperState(init_state);
+
+    // create a subscriber to the gripper's properties
+    sub_prop = rnh.subscribe("/robot/end_effector/" + _limb + "_gripper/properties",
+                                SUBSCRIBER_BUFFER, &Gripper::gripperPropCb, this);
+
+    // sleep to wait for the publisher to be ready
+    ros::Duration(0.5).sleep();
```

Comment:
note that the sleep time had to be increased from 0.05 to 0.5 so that `setParameters` on line 40 would take action.